llvm/test/CodeGen/AArch64/naked-fn-with-frame-pointer.ll

```diff
@@ -0,0 +1,11 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc < %s -mtriple aarch64 | FileCheck %s -check-prefixes=NAKED
+
+define dso_local void @_start() naked "frame-pointer"="all" {
+; NAKED-LABEL: _start:
+; NAKED:       // %bb.0:
+; NAKED-NEXT:    mov w8, #42 // =0x2a
+; NAKED-NEXT:    blr x8
+  call void inttoptr (i32 42 to ptr)()
```

Comment:
Is it legal to have non-inline-assembly statements in a naked function? 

For example clang gives me the following (https://godbolt.org/z/ozqMMWqqM)
```
__attribute__((naked)) void foo(uint64_t value) {
    constexpr int i = 1;
    __asm__("eor r0, r0, %0"::"i"(i));

}
```
`error: non-ASM statement in naked function is not supported`

GCC allows a lot more in naked functions, but maybe this test should just be an empty inline asm block?

Reply:
The LLVM language reference doesn't disallow it; it only makes mention of not emitting a prologue/epilogue.

I do strongly agree that it's a bad idea to put non-`asm` code in a naked function *in general*, but in this case, it was the only thing I could think of to write that works across all the targets we're testing here. At least ARC and SPIR-V don't like inline `asm`, for example.

Reply:
Ah fair enough, I guess we could also just do something like `call void @llvm.trap() `.

Do you think it would make sense to show the code generation for the same function without the naked attribute?

Reply:
I actually tried `@llvm.trap` first, but a few targets don't like that one either. 😅 

Is there a convenient way to show the non-`naked` case other than just duplicating the function?

Reply:
You could add another RUN line that uses `sed` to remove the naked attribute.

```
; RUN: sed 's/() naked /()/'< %s | llc -mtriple aarch64 | FileCheck %s -check-prefixes=NAKED
```

But since the function body is just two lines I'm not sure how much better that is.